File: mobile/listener_app/lib/models/event.dart
class Event {
  Event({
    required this.id,
    required this.name,
    this.description,
    this.startTime,
    this.endTime,
    required this.status,
  });

  final String id;
  final String name;
  final String? description;
  final DateTime? startTime;
  final DateTime? endTime;
  final String status;

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] as String,
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
      startTime: json['startTime'] != null ? DateTime.parse(json['startTime'] as String) : null,
      endTime: json['endTime'] != null ? DateTime.parse(json['endTime'] as String) : null,
      status: json['status'] as String? ?? 'Unknown',
    );
  }
}

File: backend/Backend.Models/Entities/Event.cs
using Backend.Models.Enums;

namespace Backend.Models.Entities;

public sealed class Event
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? StartTime { get; set; }
    public DateTime? EndTime { get; set; }
    public EventStatus Status { get; set; }
    public Guid CreatedByUserId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User CreatedByUser { get; set; } = null!;
    public ICollection<Channel> Channels { get; set; } = new List<Channel>();
    public ICollection<Session> Sessions { get; set; } = new List<Session>();
}
